build/common.sh

```diff
@@ -628,11 +632,15 @@ function kube::build::rsync_probe {
 # This will set the global var KUBE_RSYNC_ADDR to the effective port that the
 # rsync daemon can be reached out.
 function kube::build::start_rsyncd_container() {
+  IPTOOL=ifconfig
+  if kube::build::has_ip ; then
+    IPTOOL="ip address"
+  fi
   kube::build::stop_rsyncd_container
   V=3 kube::log::status "Starting rsyncd container"
   kube::build::run_build_command_ex \
     "${KUBE_RSYNC_CONTAINER_NAME}" -p 127.0.0.1:${KUBE_RSYNC_PORT}:${KUBE_CONTAINER_RSYNC_PORT} -d \
-    -e ALLOW_HOST="$(ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1')" \
+    -e ALLOW_HOST="$(${IPTOOL} | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1')" \
```

Comment:
Perhaps IP detection should be offloaded into it's own function (with local vars), and then called from here, rather than complicating this bit with another global environment variable.

Reply:
This pr makes the state of the world no worse IMO.

I think refactoring this should be a separate pr from fixing its reliance on a deprecated tool. ifconfig is deprecated at this point.

Reply:
I think a separate PR makes sense to improve the function, but this PR is fixing a build regression introduced recently.

Reply:
It does make it worse if we blindly trust that a binary named "ip" is preferred over ifconfig. If we add extra validation that "ip" is actually the iproute2 tool, then I'm okay with this. Refactor would also be a good idea, but it can be a separate PR if we validate the binary.